src/main/java/code/practice/tasks/RoomService.java

```diff
@@ -0,0 +1,83 @@
+package code.practice.tasks;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+
+public class RoomService {
+
+    //Declare a Collection to store Room Inventory
+    private Collection<Room> inventory;
+
+    //Initialize Collection and assign it to the RoomInventory
+    public RoomService() {
+        this.inventory = new LinkedHashSet<>();
+    }
+
+    //Return the Room Inventory
+    public Collection<Room> getInventory() {
+        return new HashSet<>(this.inventory);
+    }
+
+    //Add a new Room to the Room Inventory using the provided parameters
+    public void createRoom(String name, String type, int capacity, double rate) {
+        this.inventory.add(new Room(name, type, capacity, rate));
+    }
+
+    //Reduces the rate of each room by the provided discount
+    public void applyDiscount(final double discount) {
+        inventory.forEach(r -> r.setRate(r.getRate() - discount));
+    }
+
+    //Returns a new collection of rooms that meet or exceed the provided capacity
+    public Collection<Room> getRoomsByCapacity(final int requiredCapacity) {
+        Collection<Room> inventoriesWithRequiredCapacity = new HashSet<>();
+        inventory.forEach(r -> {
+            if (r.getCapacity() >= requiredCapacity) {
+                inventoriesWithRequiredCapacity.add(r);
+            }
+        });
+        return inventoriesWithRequiredCapacity;
+    }
+
+    //Returns a new collection of rooms with a rate below the provided rate and that match the provided type
+    public Collection<Room> getRoomByRateAndType(final double rate, final String type) {
+        Collection<Room> inventoriesWithProvidedRateAndType = new HashSet<>();
+        inventory.forEach(r -> {
+            if ((r.getRate() == rate) & (r.getType().equals(type))) {
+                inventoriesWithProvidedRateAndType.add(r);
+            }
+        });
+        return inventoriesWithProvidedRateAndType;
+    }
+
+    public void createRooms(Room[] rooms) {
+        this.inventory.addAll(Arrays.asList(rooms));
```

Comment:
Please use one code style in all this class, sometimes you use `this.` before using collection and sometimes you don't. I'd prefer do not use `this` word here, but it's up to you.